target/linux/ar71xx/image/tp-link.mk

```diff
@@ -1093,8 +1097,9 @@ define Device/tl-wr942n-v1
   IMAGE_SIZE := 14464k
   KERNEL := kernel-bin | patch-cmdline | lzma | uImageArcher lzma
   IMAGES := sysupgrade.bin factory.bin
-  IMAGE/sysupgrade.bin := append-rootfs | tplink-safeloader sysupgrade
+  IMAGE/sysupgrade.bin := append-rootfs | tplink-safeloader sysupgrade | append-metadata
```

Comment:
Same here, please add `check-size $$$$(IMAGE_SIZE)` as a last build step for sysupgrade image.